src/pretix/control/templates/pretixcontrol/auth/login.html

```diff
@@ -39,7 +39,7 @@
                     </a>
                 {% endif %}
                 {% if can_register %}
-                    <a href="{% url "control:auth.register" %}" class="btn btn-link btn-block">
+                    <a href="{% url "control:auth.register" %}?next={{ request.GET.next|urlencode }}" class="btn btn-link btn-block">
```

Comment:
Better split this to a function to check `next` before appending query string.
- If `next` doesn't exist, don't append query string.
- If `next` is empty string, don't append query string.